### PR TITLE
Events listing page

### LINKED ISSUE #1892: Add JSON-LD to /events
https://github.com/wellcometrust/wellcomecollection.org/pull/1890/files#r154340124

### PR DESCRIPTION
**Pull request by davidpmccormick**
References #1796
Closes #1892 

## Value
- adds starter event listing page
- adds event unstyled event information to event page template

## Screenshot
`events/`
screen shot 2017-11-30 at 18 40 18

`events/:id`
localhost_3002_ev_wcpx8ygaah4q9wgn

### CODE REVIEW

File: events/app/package.json [hunk 1]
```diff
@@ -24,5 +23,8 @@
       "babel-register"
     ],
     "babel": "inherit"
+  },
+  "devDependencies": {
+    "babel-plugin-transform-flow-strip-types": "^6.22.0"
```
Comment (jamesgorrie) [inline events/app/package.json]: This will be needed in prod too.

File: events/app/views/pages/event.njk [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  {# Contributors #}
+  {% for contributor in event.contributors %}
+    <img src="{{ contributor.person.image | convertImageUri(300) }}" alt="" />
+    <p>{{ contributor.person.name }}</p>
+    <span>{{ contributor.role.title }}</span>
+    <p>{{ contributor.person.description | safe }}</p>
+  {% endfor %}
+</article>
+{% endblock %}
```
Comment (jamesgorrie) [inline events/app/views/pages/event.njk]: FYI - I am going to override this.

File: events/app/views/pages/events.njk [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{% extends 'layout/default.njk' %}
+
+{% block pageMeta %}
+  {% set metaContent = {
+    type: 'website',
+    title: 'Events',
+    image: allEvents[0].image.contentUrl,
+    url: pageConfig.canonicalUri,
+    description: 'Past and present events at Wellcome Collection'
```
Comment (jamesgorrie) [inline events/app/views/pages/events.njk]: Let's try and get this written - not blocking, but good if we could.
@jennpb?

File: package.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
   },
   "homepage": "https://github.com/wellcometrust/wellcomecollection.org#readme",
   "devDependencies": {
+    "babel-cli": "^6.26.0",
```
Comment (jamesgorrie) [inline package.json]: Why is this needed here again?

File: server/index.js [hunk 5]
```diff
@@ -1,21 +1,27 @@
...
 export const filters = filtersMap.toJS();
 export const model = {createPageConfig};
 export const prismicParsers = {
   parsePromoListItem,
-  parseExhibitionsDoc
+  parseExhibitionsDoc,
+  prismicImage,
+  asText
 };
+export {Prismic};
```
Comment (jamesgorrie) [inline server/index.js]: See here:
https://github.com/wellcometrust/wellcomecollection.org/pull/1879#discussion_r154101728

File: events/app/views/pages/events.njk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  {% component 'open-graph', { pageConfig: pageConfig, metaContent: metaContent } %}
+  {% component 'twitter-card', { pageConfig: pageConfig, metaContent: metaContent } %}
+
+ <meta name="description" content="{{ metaContent.description }}" />
+{% endblock %}
+
+{% block body %}
+
+  <script type="application/ld+json">
+    {# TODO: add JSON+LD #}
```
Comment (jamesgorrie) [inline events/app/views/pages/events.njk]: Could this be part of the PR given that SEO is P1?
Or definitely the next step?

Comment (jamesgorrie) [reply events/app/views/pages/events.njk]: https://github.com/wellcometrust/wellcomecollection.org/issues/1892

File: events/app/views/components/event-promo/event-promo.njk [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{% set sizes = data.sizes if data.sizes else '100vw' %}
```
Comment (jamesgorrie) [inline events/app/views/components/event-promo/event-promo.njk]: Where is this from? I can't see where data is passed in?

File: events/app/views/components/event-promo/event-promo.njk [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+<a id="{{ model.id }}" href="{{ model.url }}" class="promo">
+  <div class="promo__image-container">
+    {% componentV2 'image', model.image, {}, {lazyload: true, sizesQueries: sizes, defaultSize: data.defaultSize} %}
+  </div>
+  <div class="promo__description font-HNL5-s">
+    <div class="promo__heading">
+      <h2 class="promo__title font-WB6-s">{{ model.title }}</h2>
+    </div>
+    <span>starts: {{ model.start }}</span>
```
Comment (jamesgorrie) [inline events/app/views/components/event-promo/event-promo.njk]: Could we wrap these in `time`s?

File: events/app/views/pages/events.njk (hunk 3 above)
Comment (jamesgorrie) [reply events/app/views/pages/events.njk]: Or can we put something really crap in here e.g. INSERT DESCRIPTION HERE, and address it when we can.

File: events/app/views/components/event-promo/event-promo.njk (hunk 7 above)
Comment (davidpmccormick) [reply events/app/views/components/event-promo/event-promo.njk]: Almost guaranteed that we'll need it (copied over from the current `promo` template), but you're right – I should take it out until such time as we're using it.

File: events/app/views/pages/event.njk (hunk 2 above)
Comment (davidpmccormick) [reply events/app/views/pages/event.njk]: Sure thing.

File: events/app/views/pages/events.njk (hunk 6 above)
Comment (davidpmccormick) [reply events/app/views/pages/events.njk]: @jamesgorrie I understood that you were tackling the Events JSON-LD stuff, so didn't want to duplicate the work. I think I got that from Slack – perhaps I misinterpreted something. I'll do it if you're not?

File: server/index.js (hunk 5 above)
Comment (davidpmccormick) [reply server/index.js]: You might have to talk me through what you're thinking here – why, e.g., are we exporting two prismic-related services as separate things? https://github.com/wellcometrust/wellcomecollection.org/blob/master/server/index.js#L20-L21

Comment (jamesgorrie) [reply server/index.js]: The reason for this I suppose is that we;re retrospectively doing it, so should be retrospectively trying to clean up to.

That's my question here - we could export the root Prismic object and do all the predicate stuff in the specific services as @gestchild and you've done, but my thinking was to create the Prismic service, that abstracts that out and exports it to the exhibition service, which then only has to deal with the `Exhibition` type.

the point of having the Prismic service, is exactly that, making it a service.

it might also land up with optimisations being made once that could propagate to it's dependants, but that's a maybe-one-day...

Comment (jamesgorrie) [reply server/index.js]: See here for instance, where an abstraction in that service could propagate elsewhere saving time...
https://github.com/wellcometrust/wellcomecollection.org/pull/1879#discussion_r154880114
